src/test/java/com/amihaiemil/eoyaml/benchmark/BenchmarkRunner.java

```diff
@@ -0,0 +1,63 @@
+package com.amihaiemil.eoyaml.benchmark;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.esotericsoftware.yamlbeans.YamlReader;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.*;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.Main;
+import org.openjdk.jmh.annotations.*;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Fork(value = 1, jvmArgs = {"-Xms2G", "-Xmx2G"})
+@Warmup(iterations = 3)
+@Measurement(iterations = 3)
+public class BenchmarkRunner {
+
+    public static void main(final String[] args) throws IOException {
+        Main.main(args);
+    }
+
+    @Setup
+    public void setup() {
+    }
+
+    @Benchmark
+    public void eoyaml() throws IOException {
+        try (final InputStream stream =
+                 new FileInputStream(new File("target/test-classes/benchmark", "benchmark.yml"))) {
+            Yaml.createYamlInput(stream, true)
+                .readYamlMapping();
```

Comment:
@portlek eo-yaml is the fastest because ``readYamlMapping()`` only reads the file's lines. There is no parsing done when calling ``readYamlMapping``. So I think, the benchmark comparison is not correct because the other libraries are probably parsing the Input directly.

Reply:
Hm, I see, I will add more benchmarks with get/set/getString/setString